src/eredis_client.erl

```diff
@@ -81,18 +81,21 @@ stop(Pid) ->
 %% gen_server callbacks
 %%====================================================================
 
-init([Host, Port, Database, Password, ReconnectSleep, ConnectTimeout, Options]) ->
+init([Host, Port, Database, Credentials0, ReconnectSleep, ConnectTimeout, Options]) ->
     Sentinel = case Host of
         "sentinel:"++MasterStr -> list_to_atom(MasterStr);
          _ -> undefined
     end,
     erlang:put(options, Options),
     process_flag(trap_exit, true),
+    %% cannot keep password wrapped, because otherwise troulbe after downgrade
+    #{username := Username, password := Password0} = eredis:get_credentials_info(Credentials0),
+    Password = eredis_secret:unwrap(Password0),
```

Comment:
Since we're not currently supporting hot upgrades and `eredis_secret` is already introduced and immutable, I think we could keep it wrapped in the state.  Unless this change is intended to be used in e4.4.

Reply:
I don't know if this change will be used on e4.4. I just kept the old behaviour here (that already had `eredis_secret`).